tests/unittests/sources/azure/test_imds.py

```diff
@@ -168,7 +167,7 @@ def test_basic(
             mock.call(
                 self.default_url,
                 timeout=self.timeout,
-                headers=self.headers,
+                headers_cb=mock.ANY,
```

Comment:
we don't want to lose coverage of validating the header's expected contents, is there a good way to restore that even if separately?